concourse/model/traits/component_descriptor.py

```diff
@@ -217,12 +228,38 @@ def inject_steps(self):
         yield self.descriptor_step
 
     def process_pipeline_args(self, pipeline_args: 'JobVariant'):
-        if pipeline_args.has_step('release'):
+        if pipeline_args.has_trait('release'):
```

Comment:
I think this should be done the other way around.

component-descriptor step should upload to snapshot-repo (if available/configured)
however, it should be up to the release-step to react upon the component-descriptor trait's configuration, wdyt?